### PR TITLE
Add Kms Should Allow * as Principal query for CloudFormation #1075

### PR DESCRIPTION
**Pull request by pedro-mimoso**
Kms Should not allow Principal parameter to be set as * #1075

### CODE REVIEW

File: assets/queries/cloudFormation/kms_should_allow_principal/metadata.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "kms_should_allow_principal",
+  "queryName": "Kms Should Allow * as Principal",
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/kms_should_allow_principal/metadata.json:3]: Change **query directory name** to: `kms_allows_wildcard_principal`
Change **id** to: `kms_allows_wildcard_principal`
Change **query name** to: `KMS allows wildcard principal`

File: assets/queries/cloudFormation/kms_should_allow_principal/metadata.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  "severity": "MEDIUM",
+  "category": null,
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/kms_should_allow_principal/metadata.json:5]: `Identity and Access Management`

File: assets/queries/cloudFormation/kms_should_allow_principal/metadata.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+  "descriptionText": "Kms Should not allow Principal parameter to be set as *",
+  "descriptionUrl": "https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-kms-key.html"
+}
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/kms_should_allow_principal/metadata.json:8]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: assets/queries/cloudFormation/kms_should_allow_principal/test/negative.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+          Principal:
+            AWS: arn:aws:iam::123456789012:user/Bob
+          Action:
+          - kms:DescribeKey
+          - kms:Encrypt
+          - kms:Decrypt
+          - kms:ReEncrypt*
+          - kms:GenerateDataKey
+          - kms:GenerateDataKeyWithoutPlaintext
+          Resource: '*'
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/kms_should_allow_principal/test/negative.yaml:47]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: assets/queries/cloudFormation/kms_should_allow_principal/test/positive.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+          Principal:
+            AWS: arn:aws:iam::123456789012:user/Bob
+          Action:
+          - kms:DescribeKey
+          - kms:Encrypt
+          - kms:Decrypt
+          - kms:ReEncrypt*
+          - kms:GenerateDataKey
+          - kms:GenerateDataKeyWithoutPlaintext
+          Resource: '*'
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/kms_should_allow_principal/test/positive.yaml:46]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]
Few things to fix.

**Comment by rogeriopeixotocx** [comment]
LGTM ✅

**Review by ruigomescx — APPROVED** [review]

**Pull request merged**